components/posts/posts.php

```diff
@@ -781,25 +781,29 @@ function mm_vc_posts() {
 		return;
 	}
 
+	$titles         = mm_get_post_titles_for_vc( 'mm-posts' );
 	$post_types     = mm_get_post_types_for_vc( 'mm-posts' );
 	$taxonomies     = mm_get_taxonomies_for_vc( 'mm-posts' );
 	$heading_levels = mm_get_heading_levels_for_vc( 'mm-posts' );
 	$image_sizes    = mm_get_image_sizes_for_vc( 'mm-posts' );
 	$templates      = mm_get_mm_posts_templates_for_vc( 'mm-posts' );
 
 	vc_map( array(
-		'name'     => __( 'Posts', 'mm-components' ),
-		'base'     => 'mm_posts',
-		'class'    => '',
-		'icon'     => MM_COMPONENTS_ASSETS_URL . 'component-icon.png',
-		'category' => __( 'Content', 'mm-components' ),
-		'params'   => array(
+		'name'              => __( 'Posts', 'mm-components' ),
+		'base'              => 'mm_posts',
+		'class'             => '',
+		'icon'              => MM_COMPONENTS_ASSETS_URL . 'component-icon.png',
+		'category'          => __( 'Content', 'mm-components' ),
+		'admin_enqueue_css' => MM_COMPONENTS_URL . '/css/post-titles-autocomplete.css',
+		'params'            => array(
 			array(
-				'type'        => 'textfield',
-				'heading'     => __( 'Post IDs', 'mm-components' ),
-				'param_name'  => 'post_ids',
-				'description' => __( 'Enter post IDs to display (separated by commas)', 'mm-components' ),
-				'value'       => '',
+				'type'        => 'autocomplete',
+				'heading'     => __( 'Post/Page Titles', 'mm-components' ),
+				'param_name'  => 'post_titles',
```

Comment:
Continuing from my previous comment, I think this can stay post_titles, and you don't have to include post_ids as a VC param here at all, but it would be ideal if both could be sent into the mm_posts function separately. In the mm_posts function I think you can first look for `post_titles`, and if that has a value use it, otherwise if `post_ids` has a value use that, so that you only send in one `post__in` arg to WP_Query.

If you want to also support passing `post_titles` as actual post titles or slugs, see the `post_name__in` arg you can pass to WP_Query as of 4.4. Looks like it only works with slugs, and I'd say it's probably always going to be a bit faster if you pass IDs, but it's there if you wanted to do it.

Another cool param to support in `mm_posts` might be `s`, which does a text search. The use case would be to show all posts that match a search term.